[inductor][addmm] incorporate into new get_mm_configs properly

coconutruben opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #161534

# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @jataylo @chenyang78

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

coconutruben committed:
[inductor][addmm] incorporate into new get_mm_configs properly

\# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

\# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

\# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

eellison reviewed (commented):
just for my understanding - how is this different than adjust_kernel_inputs? adjust_kernel_inputs should only be applying views, right ?

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- expand KernelInputs to also store views of specific nodes, by names
- use that view (inp, the unexpanded version) in the heuristics to
  adjust it depending on whether we're in max-autotune or not

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- remove the view from inp_expanded when running not in max-autotune

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- remove the view from inp_expanded when running not in max-autotune

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben commented:
@coconutruben has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D81520581).

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

jansel reviewed (changes requested):
Failing tests?

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]

coconutruben committed:
Update on "[inductor][addmm] incorporate into new get_mm_configs properly"


# why

- addmm aten running with an expanded version of bias vs the regular
  bias sometimes causes numerics differences
- to avoid this for now, we make addmm aten use inp vs inp_expanded
  depending on if we're in max-autotune or not, matching the previous
  logic

# what

- pass unexpanded bias (inp)
- let template (heuristics) that it to be expanded (ATen in not max-autotune, Triton always) expand it

# testing

```
python3 -bb -m pytest test/inductor/test_aot_inductor.py::AOTInductorTestABICompatibleCpu::test_aoti_debug_printer_codegen_cpu
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D81520581](https://our.internmc.facebook.com/intern/diff/D81520581)

[ghstack-poisoned]
